src/comparator/file_set_comparator.py

```diff
@@ -0,0 +1,84 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from google.protobuf.descriptor_pb2 import FileDescriptorSet
+from google.protobuf.descriptor_pb2 import FileOptions
+from google.protobuf.descriptor_pb2 import DescriptorProto
+from google.protobuf.descriptor_pb2 import ServiceDescriptorProto
+from google.protobuf.descriptor_pb2 import EnumDescriptorProto
+from src.comparator.service_comparator import ServiceComparator
+from src.findings.finding_container import FindingContainer
+from src.findings.utils import FindingCategory
+from typing import Dict, Optional
+
+
+class _FileSet:
+    # TODO(xiaozhenliu): check with One-platform about the version naming.
+    # We should allow minor version updates, then the packaging options like
+    # `java_package = "com.pubsub.v1"` will always be changed. But versions
+    # updates between two stable versions (e.g. v1 to v2) is not permitted.
+
+    def __init__(self, file_set: FileDescriptorSet):
+        self.packaging_options_map = {}
+        self.services_map: Dict[str, ServiceDescriptorProto] = {}
+        self.messages_map: Dict[str, DescriptorProto] = {}
+        self.enums_map: Dict[str, EnumDescriptorProto] = {}
+        for fd in file_set.file:
+            # Create packaging options map and duplicate the per-language rules for namespaces.
+            self.packaging_options_map = self._get_packaging_options_map(fd.options)
+            for service in fd.service:
+                self.services_map[service.name] = service
```

Comment:
Personal preference nit, feel free to ignore: dicts can be updated in bulk with the `update` method.
```python
self.services_map.update((service.name, service) for service in fd.service)
```

Reply:
Thanks, this looks cleaner!